src/testing/run.ts

```diff
@@ -145,9 +146,17 @@ async function runTestCaseUnsafe<TestCaseType, OutputType>(args: {
     throw new Error(`[${args.testId}] Test case semaphore not found.`);
   }
 
-  const output = await semaphore.run(async () => {
-    return await args.fn({ testCase: args.testCase });
-  });
+  const output = await testCaseRunAsyncLocalStorage.run(
+    {
+      testCaseHash: args.testCaseHash,
+      testId: args.testId,
+    },
+    async () => {
+      return await semaphore.run(async () => {
+        return await args.fn({ testCase: args.testCase });
+      });
+    },
+  );
```

Comment:
Unexpectedly if I change the order of testCaseRunAsyncLocalStorage.run and semaphore.run, then the concurrency test fails 🤔

Reply:
hmm